src/delimited-stream.ts

```diff
@@ -0,0 +1,30 @@
+import StreamSearch = require('streamsearch');
+import { PromiseNext } from './promise-next';
+
+export type ReadCallback = (chunk: ArrayBuffer | SharedArrayBuffer) => any;
+
+export interface Readable {
+  on(event: 'data', callback: ReadCallback): any;
+}
+
+export type NextBuffer = PromiseNext<Buffer>;
+
+export function delimit(inputStream: Readable, delimiter: string): NextBuffer {
+  const searcher = new StreamSearch(delimiter);
+
+  const ret = new PromiseNext<Buffer>();
+
+  let bufs: Buffer[] = [];
+  searcher.on('info', (isMatch: boolean, data: Buffer, start: number, end: number) => {
+    if (data) bufs.push(data.subarray(start, end));
```

Comment:
Lint says `This type is not allowed in the 'if' condition because it is always truthy. Allowed types are boolean, null-union, undefined-union, string, or number.`, please ensure lint after this PR.